### PR TITLE
Doc fixes

### LINKED ISSUE #1586: Add missing doc links
Some of the docs are referenced and thus hard to find.

### PR DESCRIPTION
**Pull request by jodh-intel**
Added links to the unreferenced docs in this repo to make them more discoverable.

Also added some missing TOCs and simplified a link.

Fixes #1586.
    
Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

### CODE REVIEW

File: cmd/checkmetrics/README.md [hunk 1]
```diff
@@ -1,5 +1,23 @@
 # checkmetrics
 
+* [checkmetrics](#checkmetrics)
```
Comment (grahamwhaley) [inline cmd/checkmetrics/README.md]: slightly confused - should this and the next patch maybe folded into one?

Comment (jodh-intel) [reply cmd/checkmetrics/README.md]: Ah - that link can in fact be zapped. But I'm not quite clear on what you mean - the next commit is for `checkcommits` ?There are a number of ways these changes could be ordered but I've tried to patch-up the existing files, then finally add in missing links.

Comment (grahamwhaley) [reply cmd/checkmetrics/README.md]: doh, I mis-read and muddle up `checkcommits` and `checkmetrics` in my head all the time - heh, and I tried to stare quite hard at this one - mea culpa - ignore me!

**Comment by chavafg** [comment]
I've slightly modified most of the jobs in jenkins to not run if only `*.md` were modified. (except the ubuntu job which is `Required`).
Forgot to add the change to the metrics and ARM job, but have already canceled the arm job manually. 

The downside I see about this approach is that if we have `Required` on all jobs (which I think we do when they are stable), the PR will be blocked to as it will not get the status from the jenkins server.

**Comment by jodh-intel** [comment]
Merging - no point waiting for a CI to test a doc-only PR...

**Pull request merged**